Sources/StevensonCore/JiraService.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+import Vapor
+
+public struct JiraService {
+    public let baseUrl: String
+    private let headers: HTTPHeaders
+
+    public init(baseUrl: String, username: String, password: String) {
+        self.baseUrl = "\(baseUrl):443"
+
+        let base64Auth = Data("\(username):\(password)".utf8).base64EncodedString(options: [])
+        self.headers = [
+            "Authorization": "Basic \(base64Auth)",
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        ]
+    }
+}
+
+// MARK: Issue creation
+
+public protocol JiraIssueFields: Content {
+    var project: JiraService.FieldType.ObjectID { get }
+    var issueType: JiraService.FieldType.ObjectID { get }
+    func makeIssue() -> JiraService.Issue<Self>
```

Comment:
can we do it another way that would not introduce "self-requirement"? Do we really need this method or maybe just `JiraService.Issue.init` would be enough? I'm not sure about the cases when this can be differently implemented.

Reply:
I tried that but failed to make the compiler happy. I tried to add the init in an `extension JiraService.Issue where Fields = CRPIssueFields` but my Xcode kept yelling and even crashing, hence the inversion of responsibility here instead 😉
If you manage to make it work on a stake way I'm interested though